dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/ClientIT.java

```diff
@@ -300,112 +348,203 @@ private void assertDepsCommandOutput() {
+        checkSuggestionIsGiven(new String[] { }, PLUGIN);
+        checkSuggestionIsGiven(new String[] { }, UPGRADE);
+        checkSuggestionIsGiven(new String[] { }, HELP);
+
+        checkSuggestionIsGiven(new String[] {TOOL}, CONVERT);
+        checkSuggestionIsGiven(new String[] {TOOL}, TEST_PARAMETER);
+        checkSuggestionIsGiven(new String[] {TOOL}, HELP);
+
+        checkSuggestionIsGiven(new String[] {WORKFLOW}, CONVERT);
+        checkSuggestionIsGiven(new String[] {WORKFLOW}, TEST_PARAMETER);
+        checkSuggestionIsGiven(new String[] {WORKFLOW}, HELP);
+
+        checkSuggestionIsGiven(new String[] {TOOL, CONVERT}, HELP);
+        checkSuggestionIsGiven(new String[] {TOOL, CONVERT}, CWL_2_JSON);
+
+        checkSuggestionIsGiven(new String[] {WORKFLOW, CONVERT}, HELP);
+        checkSuggestionIsGiven(new String[] {WORKFLOW, CONVERT}, CWL_2_JSON);
+
+        checkSuggestionIsGiven(new String[] {WORKFLOW, WES}, LAUNCH);
+        checkSuggestionIsGiven(new String[] {WORKFLOW, WES}, STATUS);
+
+        checkSuggestionIsGiven(new String[] {WORKFLOW, WES, LAUNCH}, INLINE_WORKFLOW);
+        checkSuggestionIsGiven(new String[] {WORKFLOW, WES, STATUS}, ID);
+        checkSuggestionIsGiven(new String[] {WORKFLOW, WES, LOGS}, ID);
+        checkSuggestionIsGiven(new String[] {WORKFLOW, WES, CANCEL}, ID);
+        checkSuggestionIsGiven(new String[] {WORKFLOW, WES, SERVICE_INFO}, WES_URL);
+        checkSuggestionIsGiven(new String[] {WORKFLOW, WES, LIST}, PAGE_TOKEN);
+
+        checkSuggestionIsGiven(new String[] {PLUGIN}, HELP);
+        checkSuggestionIsGiven(new String[] {PLUGIN}, LIST);
+        checkSuggestionIsGiven(new String[] {PLUGIN}, DOWNLOAD);
+
+        checkSuggestionIsGiven(new String[] {DEPS}, CLIENT_VERSION);
+        checkSuggestionIsGiven(new String[] {DEPS}, HELP);
+
+        checkSuggestionIsGiven(new String[] {CHECKER}, HELP);
+        checkSuggestionIsGiven(new String[] {CHECKER}, DOWNLOAD);
+
+    }
+
+    /** This method performs two tests, in the first test commandToBeTested is capitalised, and it ensures that
+     * commandToBeTested is given as a suggestion. In the second test, the first letter of commandToBeTested is changed,
+     * and it ensures that commandToBeTested is given as a suggestion.
+     *
+     * @param validCommands A list of commands that are valid (these commands will not be modified by this method)
+     * @param commandToBeTested A command that is valid, but will be modified to become invalid by this method
+     */
+    private void checkSuggestionIsGiven(String[] validCommands, String commandToBeTested) throws IOException {
+        String formmatedValidCommandString = "";
+        for (String command: validCommands) {
+            formmatedValidCommandString += " ";
+            formmatedValidCommandString += command;
+        }
+
+        final ArrayList<String> commandsForCapitalizationTest = Lists.newArrayList(validCommands);
```

Comment:
Probably doesn't need to be an `ArrayList` variable, if you use `List` it is less typing but also emphasizes the implementation is irrelevant

Reply:
Fixed!